src/db/tables/subscriber_coupons.ts

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import createDbConnection from "../connect.js";
+import { logger } from "../../app/functions/server/logging";
+import { SubscriberCouponRow } from "knex/types/tables";
+
+const knex = createDbConnection();
+
+async function checkCouponForSubscriber(
+  subscriberId: number,
+  couponCode: string,
+) {
+  logger.info("checkCouponForSubscriber", subscriberId);
+  return (
+    (
+      await knex("subscriber_coupons")
+        .orderBy("created_at")
+        .where("subscriber_id", subscriberId)
+        .andWhere("coupon_code", couponCode)
+        .returning("*")
+    ).length > 0
```

Comment:
Hm, took me a moment to realize this was an existence check... if we use raw you could use [SELECT EXISTS](https://www.postgresql.org/docs/current/functions-subquery.html#FUNCTIONS-SUBQUERY-EXISTS) but up to you. I wish `knex` just had a `.exists()`!

Reply:
I rewrote it in a simpler way without using raw: https://github.com/mozilla/blurts-server/pull/4633/commits/adc67e8645c9cc49b674fbce25b4fdbeba8dd7bd